x/nexus/keeper/chain.go

```diff
@@ -248,3 +223,75 @@ func (k Keeper) setChainByNativeAsset(ctx sdk.Context, asset string, chain expor
 func (k Keeper) GetChainByNativeAsset(ctx sdk.Context, asset string) (chain exported.Chain, ok bool) {
 	return chain, k.getStore(ctx).Get(chainByNativeAssetPrefix.Append(utils.LowerCaseKey(asset)), &chain)
 }
+
+type ChainKeeper struct {
+	k      Keeper
+	states map[string]*cached.Cached[types.ChainState]
+	dirty  map[string]bool
```

Comment:
Not quite sure why we need to allow caching all chain states. Since each chain state needs to be updated independently, it seems cleaner to wrap a single chain state. Maybe nexus keeper can be queried to retrieve a caching chain state given a chain name. We can mark vote counts etc. and call persist at the end